src/servant/tt_002/3678.c

```diff
@@ -36,7 +44,101 @@ INCLUDE_ASM("servant/tt_002/nonmatchings/3678", func_us_80173BD0);
 
 INCLUDE_ASM("servant/tt_002/nonmatchings/3678", func_us_80173D60);
 
-INCLUDE_ASM("servant/tt_002/nonmatchings/3678", func_us_801746E8);
+void func_us_801746E8(InitializeMode mode) {
+    u16* src;
+    u16* dst;
+    RECT rect;
+    s32 i;
+    SpriteParts** spriteBanks;
+    Entity* entity;
+
+    s_ServantId = g_Servant;
+
+    if ((mode == MENU_SWITCH_SERVANT) || (mode == MENU_SAME_SERVANT)) {
+        ProcessEvent(NULL, true);
+        if (mode == MENU_SAME_SERVANT) {
+            return;
+        }
+    }
+
+    dst = &g_Clut[CLUT_INDEX_SERVANT];
+    src = D_us_801722E8; // clut data for faerie, will rename at data import
+    for (i = 0; i < 0x100; i++) {
+        *dst++ = *src++;
+    }
+
+    rect.x = 0;
+    rect.w = 0x100;
+    rect.h = 1;
+    rect.y = 0xF4;
+
+    dst = &g_Clut[CLUT_INDEX_SERVANT];
+    LoadImage(&rect, (u_long*)dst);
+
+    spriteBanks = g_api.o.spriteBanks;
+    spriteBanks += 20;
+    *spriteBanks = (SpriteParts*)g_ServantSpriteParts;
+
+    entity = &thisFamiliar;
+
+    DestroyEntity(entity);
+    entity->unk5A = 0x6C;
+    entity->palette = 0x140;
+    entity->animSet = ANIMSET_OVL(20);
+    entity->zPriority = PLAYER.zPriority - 2;
+    entity->facingLeft = PLAYER.facingLeft;
+    entity->params = 0;
+
+    if (mode == MENU_SWITCH_SERVANT) {
+
+// PSP version does this in 2 chunks, the PSX version uses an lw instruction
+#ifdef VERSION_PSP
+        if (D_8003C708.flags & LAYOUT_RECT_PARAMS_UNKNOWN_20 ||
+            D_8003C708.flags & LAYOUT_RECT_PARAMS_UNKNOWN_40) {
+#else
+        if (LOW(D_8003C708.flags) &
```

Comment:
I suspect this is something like `if ((D_8003C708.flags & 0x60) && (D_8003C708.unk2 == 0)) {` but can't seem to get the optimization to trigger